demisto_sdk/commands/setup_env/setup_environment.py

```diff
@@ -0,0 +1,406 @@
+from demisto_sdk.commands.common.handlers import JSON_Handler
+from demisto_sdk.commands.common.handlers.json.json5_handler import JSON5_Handler
+from demisto_sdk.commands.common.logger import logger
+from demisto_sdk.commands.content_graph.objects.base_content import BaseContent
+from demisto_sdk.commands.content_graph.objects.integration import Integration
+from demisto_sdk.commands.content_graph.objects.integration_script import (
+    IntegrationScript,
+)
+from demisto_sdk.commands.content_graph.objects.pack import Pack
+from demisto_sdk.commands.setup_env.configure_integration_in_server import (
+    create_integration_instance,
+)
+
+json5 = JSON5_Handler()
+json = JSON_Handler()
+
+
+class IDE(Enum):
+    VSCODE = "vscode"
+    PYCHARM = "pycharm"
+
+
+IDE_TO_FOLDER = {IDE.VSCODE: ".vscode", IDE.PYCHARM: ".idea"}
+
+
+def get_integration_params(project_id: str, secret_id: str) -> dict:
+    """This function retrieves the parameters of an integration from Google Secret Manager
+    *Note*: This function will not run if the `DEMISTO_SDK_GCP_PROJECT_ID` env variable is not set.
+
+    Args:
+        project_id (str): GSM project id
+        secret_id (str): The secret id in GSM
+
+    Returns:
+        dict: The integration params
+    """
+    # Create the Secret Manager client.
+    client = secretmanager.SecretManagerServiceClient()
+
+    # Build the resource name of the secret version.
+    name = f"projects/{project_id}/secrets/{secret_id}/versions/latest"
+
+    # Access the secret version.
+    try:
+        response = client.access_secret_version(name=name)
+    except google.api_core.exceptions.NotFound:
+        logger.warning("The secret is not found in the secret manager")
+        return {}
+    except google.api_core.exceptions.PermissionDenied:
+        logger.warning(
+            "Insufficient permissions for gcloud. If you have the correct permissions, run `gcloud auth application-default login`"
+        )
+        return {}
+    except Exception:
+        logger.warning(
+            f"Failed to get secret {secret_id} from secret manager, skipping"
+        )
+        return {}
+    # Return the decoded payload.
+    payload = response.payload.data.decode("UTF-8")
+    return json5.loads(payload).get("params")
+
+
+def add_init_file_in_test_data(integration_script: IntegrationScript):
+    if (integration_script.path.parent / "test_data").exists():
+        (integration_script.path.parent / "test_data" / "__init__.py").touch()
+
+
+def configure_dotenv():
+    """
+    This functions configures the .env file located with PYTHONPATH and MYPYPATH
+    """
+    dotenv_path = CONTENT_PATH / ".env"
+    env_vars = dotenv.dotenv_values(dotenv_path)
+    env_vars["PYTHONPATH"] = ":".join([str(path) for path in PYTHONPATH])
+    env_vars["MYPYPATH"] = ":".join([str(path) for path in PYTHONPATH])
+    for key, value in env_vars.items():
+        if not value:
+            continue
+        dotenv.set_key(dotenv_path, key, value)
```

Comment:
```suggestion
        if value:
            dotenv.set_key(dotenv_path, key, value)
        else:
            logger.warning(f"empty value for {key}, not setting it")
```

Reply:
P.S why would we ever have empty values?

Reply:
It can happan if a user configured on his own `.env` file an empty env, like "DEMISTO=". This functions adds to the env file, it doesn't overwrite existing values.